internal/controller/appconfigurationprovider_controller.go

```diff
@@ -269,15 +269,15 @@ func (reconciler *AzureAppConfigurationProviderReconciler) Reconcile(ctx context
 			}
```

Comment:
Here is the right place to check `shouldCreateOrUpdate`

```
for name := range processor.Settings.SecretSettings {
     // check the target object existence

    if (shouldCreateOrUpdate){
               result, err := reconciler.createOrUpdateSecret(ctx, provider, SecretSetting)
		if err != nil {
			return result, nil
		}
    }

}
```

Reply:
We'd better not let one method to perform all secrets update. We check and update for each.